cfme/utils/ssh.py

```diff
@@ -276,6 +276,18 @@ def run_command(
         Returns:
             A :py:class:`SSHResult` instance.
         """
+        # paramiko hangs on *_ready calls if destination has become unavailable
+        # this is some kind of watchdog to handle this issue
+        try:
+            with gevent.Timeout(timeout):
+                return self._run_command(command, timeout, reraise, ensure_host, ensure_user,
```

Comment:
**QUESTION**
I just wonder if this warning applicable here:

>    Timeouts can only work when the greenlet switches to the hub.
>    If a blocking function is called or an intense calculation is ongoing during
>    which no switches occur, :class:`Timeout` is powerless.
http://www.gevent.org/_modules/gevent/timeout.html

Reply:
your concern is right. But it is not relevant here. 
the function which hangs does something. it allows to switch contexts. actually, it indefinitely polls socket.

I tested my changes in a real situation.
this with statement is just short form of 
```
gevent.spawn(command)
gevent.spawn_later(timeout, kill_command)
```